src/Ecomm.php

```diff
@@ -184,8 +195,37 @@ public function createRecurringPayment($amount, $description, $expiry, $language
         return $this->sendRequest($params);
     }
 
+    /**
+     * @param $amount
+     * @param $description
+     * @param $expiry
+     * @param $language
+     * @return array
+     * @throws EcommException
+     */
+    public function createDMSAddCardRequest($amount, $description, $expiry, $language = 'en')
+    {
+        $params = [
+            'command' => 'd',
+            'amount' => $amount,
+            'currency' => $this->currency,
+            'client_ip_addr' => $this->client_ip_addr,
+            'description' => $description,
+            'language' => $language,
+            'msg_type' => 'DMS',
+            'biller_client_id' => Uuid::uuid4(),
+            'perspayee_expiry' => date('my', strtotime($expiry)),
+            'perspayee_gen' => '1',
+            'oneclick' => 'Y',
+        ];
+        return $this->sendRequest($params);
+    }
+
     public function createAuthorizationRequest($amount, $description, $cardReference, $language = 'en')
     {
+        if(!$this->useDMS()) {
+            throw new NotSupportedException('According to settings pre-authorization terminal is not supported');
```

Comment:
I'm afraid no one handles this, I think we want to make sure we mark our internal transaction as failed with some reason that gives info about the misconfiguration